r/tools/linuxlibs.R

```diff
@@ -233,7 +240,7 @@ ensure_cmake <- function() {
 }
 
 # TODO: move ensure_flex/bison/m4 to cmake: https://issues.apache.org/jira/browse/ARROW-7501
-ensure_flex <- function(m4 = ensure_m4()) {
+ensure_flex <- function(m4 = ensure_m4(), bison = ensure_bison()) {
```

Comment:
Well, yes, Flex fails building otherwise.

Reply:
I of course would be happy to have this bison/flex code moved out of this R script and into cmake proper (https://issues.apache.org/jira/browse/ARROW-7501).

Reply:
I'm only interested in fixing this and speeding it up  :)

Reply:
How did you get Flex to fail to build otherwise? Maybe we should add that container to our build matrix. It's not failing for crossbow or GHA, and I never had it fail locally for me with those docker images (though I did see the message that flex was building without bison and chose not to care since thrift successfully built).

Reply:
I ran the container locally, roughly like this:
```
$ export R_TAG=3.6-centos7
$ export R_IMAGE=r-base
$ export R_ORG=rstudio
$ docker-compose build r
$ docker-compose run --rm r
```
It's the same container as on Travis-CI, AFAICT.

Reply:
Yeah that's what's running on all commits and it's passing on master. It makes some noise about flex not having bison but it says not to worry: https://github.com/apache/arrow/runs/405798341#step:5:809

```
checking for bison... no
configure: no bison program found: only required for maintainers
```

I'm fine with the change if everything passes, but I suspect it's not necessary.

Reply:
Ah, I see... I saw the same error as here and thought that was the cause for the failing builds:
https://github.com/apache/arrow/runs/405798341#step:5:1028

I could try to remove those changes.